rpc/wrtc_client_channel.go

```diff
@@ -336,29 +333,44 @@ func (ch *webrtcClientChannel) writeReset(stream *webrtcpb.Stream) error {
 
 // taken from
 // https://github.com/grpc-ecosystem/go-grpc-middleware/blob/560829fc74fcf9a69b7ab01d484f8b8961dc734b/logging/zap/client_interceptors.go
-
-func logFinalClientLine(ctx context.Context, startTime time.Time, err error, msg string) {
+func logFinalClientLine(logger utils.ZapCompatibleLogger, startTime time.Time, err error, msg string) {
 	code := grpc_logging.DefaultErrorToCode(err)
 	level := grpc_zap.DefaultCodeToLevel(code)
+
+	// this calculation is done because duration.Milliseconds() will return an integer, which is not precise enough.
+	duration := float32(time.Since(startTime).Nanoseconds()/1000) / 1000
+	fields := []any{}
 	if err == nil {
 		level = zap.DebugLevel
+	} else {
+		fields = append(fields, "error", err)
+	}
+	fields = append(fields, "grpc.code", code.String(), "grpc.time_ms", duration)
+
+	if logger.Level().Enabled(level) {
+		// cases for all outputs of grpc_zap.DefaultCodeToLevel
+		switch level {
+		case zap.DebugLevel:
+			logger.Debugw(msg, fields...)
+		case zap.InfoLevel:
+			logger.Infow(msg, fields...)
+		case zap.ErrorLevel:
+			logger.Errorw(msg, fields...)
+		case zap.WarnLevel:
+			fallthrough
+		default:
+			logger.Warnw(msg, fields...)
+		}
```

Comment:
this is pretty ugly, but the `Write()` methods differ between zap loggers and rdk loggers so I could write another conversion method like `AddFieldsToLogger` or do this